Core/include/Acts/Propagator/EigenStepper.hpp

```diff
@@ -213,17 +213,24 @@ class EigenStepper {
     return state.pars.template segment<3>(eFreeDir0);
   }
 
+  /// QoP direction accessor
+  ///
+  /// @param state [in] The stepping state (thread-local cache)
+  double qop(const State& state) const { return state.pars[eFreeQOverP]; }
+
   /// Absolute momentum accessor
   ///
   /// @param state [in] The stepping state (thread-local cache)
   double momentum(const State& state) const {
-    return std::abs((state.q == 0. ? 1. : state.q) / state.pars[eFreeQOverP]);
+    return std::abs(charge(state) / qop(state));
```

Comment:
What happens if `q==0`?

Reply:
Charge will be `+-0`, but you take `abs`, so it's `|p|`, which is all good.

Reply:
oh yeah I have to catch the q=0 case